Px.Utils/Validation/SyntaxValidation/SyntaxValidation.cs

```diff
@@ -0,0 +1,341 @@
+        {
+            foreach (var @object in objects)
+            {
+                foreach (ValidationFunctionDelegate function in validationFunctions)
+                {
+                    ValidationFeedbackItem? feedback = function(@object, syntaxConf);
+                    if (feedback is not null)
+                    {
+                        validationFeedback.Add((ValidationFeedbackItem)feedback);
+                    }
+                }
+            }
+        }
+
+        private static List<ValidationKeyValuePair> BuildKeyValuePairs(List<ValidationEntry> validationEntries, PxFileSyntaxConf syntaxConf)
+        {
+            return validationEntries.Select(entry =>
+            {
+                // Split the entry string into a key and a value from the first valid keyword separator
+                int[] keywordSeparatorIndeces = SyntaxValidationUtilityMethods.FindKeywordSeparatorIndeces(entry.EntryString, syntaxConf);
+
+                // Key is the part of the entry string before the first keyword separator index
+                string key = entry.EntryString[..keywordSeparatorIndeces[0]];
+                // Remove preceding linebreaks and carriage returns from the key
+                key = key
+                    .TrimStart(syntaxConf.Symbols.CarriageReturn)
+                    .TrimStart(syntaxConf.Symbols.Linebreak);
+                // Value is the part of the entry string after the first keyword separator index
+                int startIndex = keywordSeparatorIndeces[0] + 1;
+                string value = entry.EntryString[startIndex..];
+
+                return new ValidationKeyValuePair(entry.Line, entry.Character, entry.File, new KeyValuePair<string, string>(key, value));
+            }).ToList();
+        }
+
+        private static List<ValidationStruct> BuildValidationStructs(List<ValidationKeyValuePair> keyValuePairs, PxFileSyntaxConf syntaxConf)
+        {
+            return keyValuePairs.Select(entry =>
+            {
+                ValidationStructKey key = ParseValidationStructKey(entry.KeyValuePair.Key, syntaxConf);
+                return new ValidationStruct(entry.Line, entry.Character, entry.File, key, entry.KeyValuePair.Value);
+            }).ToList();
+        }
+
+        private static List<ValidationEntry> BuildValidationEntries(Stream stream, Encoding encoding, PxFileSyntaxConf syntaxConf, string filename, int bufferSize)
+        {
+            int line = 1;
+            int character = 0;
+            bool isProcessingString = false;
+
+            using StreamReader reader = new(stream, encoding);
+            List<ValidationEntry> entries = [];
+            StringBuilder entryBuilder = new();
+            char[] buffer = new char[bufferSize];
+
+            while (reader.Read(buffer, 0, bufferSize) > 0)
+            {
+                for (int i = 0; i < buffer.Length; i++)
```

Comment:
The for loop should stop at the number the read method returned, not the whole buffer length.